First commits

mcdonnnj opened the pull request:
# <!-- Use the title to describe PR changes in the imperative mood --> #

## 🗣 Description ##

This initial pull request adds a workflow to represent the current `lint` job from [cisagov/skeleton-generic](https://github.com/cisagov/skeleton-generic) (and subsequently by all of our repositories that use the [Lineage](https://github.com/cisagov/action-lineage) system).
<!-- Describe the "what" of your changes in detail. -->
<!-- To avoid scope creep, limit changes to a single goal. -->

## 💭 Motivation and context ##

Migrating our GitHub Actions tasks that are common across repositories to [reusable workflows](https://docs.github.com/en/actions/using-workflows/reusing-workflows) is a maintenance and updating win. Once our other repositories are configured to use the `common-lint.yml` workflow any updates to the workflow here will be available without substantial modification to our other projects using this workflow.
<!-- Why is this change required? -->
<!-- What problem does this change solve? How did you solve it? -->
<!-- Mention any related issue(s) here using appropriate keywords such -->
<!-- as "closes" or "resolves" to auto-close them on merge. -->

## 🧪 Testing ##

Automated tests pass. I created a [branch in cisagov/skeleton-generic](https://github.com/cisagov/skeleton-generic/tree/testing/lint_job_as_reusable_workflow) to test this and the [Action run](https://github.com/cisagov/skeleton-generic/actions/runs/3513124963) ran successfully using the workflow added in this pull request.
<!-- How did you test your changes? How could someone else test this PR? -->
<!-- Include details of your testing environment, and the tests you ran to -->
<!-- see how your change affects other areas of the code, etc. -->

<!--
## 📷 Screenshots (if appropriate) ##

Uncomment this section if a screenshot is needed.

-->

## ✅ Pre-approval checklist ##

<!-- Remove any of the following that do not apply. -->
<!-- Draft PRs should have one or more unchecked boxes. -->
<!-- If you're unsure about any of these, don't hesitate to ask. -->
<!-- We're here to help! -->

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All relevant repo and/or project documentation has been updated
      to reflect the changes in this PR.
- [x] All new and existing tests pass.

## ✅ Post-merge checklist ##

<!-- Remove any of the following that do not apply. -->

- [ ] Add a tag or create a release.

mcdonnnj committed:
Add the `lint` GitHub Actions job as a reusable workflow

Convert the `lint` job from the GitHub Actions `build.yml` workflow
into a reusable workflow. This will allow all repositories in the org
to use this workflow without needing to define it in each repository.
A number of inputs have been added to ensure the workflow has enough
flexibility since it is no longer defined in a given repository.

mcdonnnj committed:
Add a `bump_version.sh` script

Add the `bump_version.sh` script from cisagov/skeleton-packer to manage
project versioning. This includes adding the `semver` package to the
development requirements to support this script.

mcdonnnj committed:
Add a file to track project versioning

Update the `bump_version.sh` script to support this file's
implementation in this project.

mcdonnnj committed:
Update the repository README

Update the README to accurately describe the goal of this
repository.

mcdonnnj committed:
Add information about the `common-lint.yml` workflow

Document the purpose and inputs/secrets for the `common-lint.yml`
workflow.

jsf9k reviewed (changes requested):
This looks great to me.  It's a great improvement and a step down the right path.  I'll be glad to see some of the GH Actions stuff DRY out a bit.

I have one (very minor) requested change:
- Document in the README why we made the decision to store multiple workflows in a single repo.

I have two questions for @team-ois's consideration:
- Is it worth documenting in the README the other workflows that the individual workflows will pull down and run?
- Even though they are small jobs, there may be some benefit to breaking out the `Install Packer`, `Install shfmt`, and `Install terraform-docs` into their own workflows (likely in separate repos) so that other non-CISA folks can easily make use of them.  That doesn't have to happen now, but if the @team-ois consensus is that it's a good idea then I can create issues.

mcdonnnj commented:
> This looks great to me. It's a great improvement and a step down the right path. I'll be glad to see some of the GH Actions stuff DRY out a bit.
> 
> I have one (very minor) requested change:
> 
>     * Document in the README why we made the decision to store multiple workflows in a single repo.
> 
> 
> I have two questions for @team-ois's consideration:
> 
>     * Is it worth documenting in the README the other workflows that the individual workflows will pull down and run?
> 
>     * Even though they are small jobs, there may be some benefit to breaking out the `Install Packer`, `Install shfmt`, and `Install terraform-docs` into their own workflows (likely in separate repos) so that other non-CISA folks can easily make use of them.  That doesn't have to happen now, but if the @team-ois consensus is that it's a good idea then I can create issues.

Regarding `Install Packer` I would much rather we finish what I started in [cisagov/setup-packer](https://github.com/cisagov/setup-packer) (forked [hashicorp/setup-terraform](https://github.com/hashicorp/setup-terraform) and started changing it to install Packer instead). I would also say that the `Install shfmt` and `Install terraform-docs` steps could be a more generic "Install Go tool" composite Action that is then used in workflows as needed.

jsf9k commented:
> Regarding `Install Packer` I would much rather we finish what I started in [cisagov/setup-packer](https://github.com/cisagov/setup-packer) (forked [hashicorp/setup-terraform](https://github.com/hashicorp/setup-terraform) and started changing it to install Packer instead). I would also say that the `Install shfmt` and `Install terraform-docs` steps could be a more generic "Install Go tool" composite Action that is then used in workflows as needed.

I'm fine with both those ideas.

dav3r commented on .github/workflows/common-lint.yml:
```diff
@@ -0,0 +1,150 @@
+          # cache because if we were caching, then we would need to use
+          # the `-upgrade=true` option. This option blindly pulls down the
+          # latest modules and providers instead of checking to see if an
+          # update is required. That behavior defeats the benefits of caching.
```
Punctuation fix:
```suggestion
          # update is required. That behavior defeats the benefits of caching,
```

dav3r reviewed (approved):
This is excellent work!  I agreement with @jsf9k's [request to document our decision to store multiple workflows in this repo](https://github.com/cisagov/github-actions-workflows/pull/14#pullrequestreview-1188503215) and with @mcdonnnj's [ideas for installing packer and go-based tools](https://github.com/cisagov/github-actions-workflows/pull/14#issuecomment-1322289879).

mcdonnnj committed:
Use the cisagov/setup-go-package action

Now that the Go package installation logic has been broken out into its
own project (cisagov/setup-go-package) we can update the steps that
install Go packages to use this action.

jasonodoom committed:
Add scorecards

jasonodoom committed:
Merge branch 'more-workflows' into first-commits

jasonodoom committed:
Add  changes to appease pre-comit

jasonodoom committed:
Add workflows

jasonodoom committed:
Fix common-lint config

jasonodoom committed:
Remove common-lint for now

jasonodoom committed:
Comment uses for common lint

jasonodoom committed:
Add common-lint

jasonodoom committed:
Added domains to list of allow list

jasonodoom committed:
Add command on new line to appease precommit

jasonodoom committed:
Specify actions version to use

mcdonnnj commented:
This branch has been changed far beyond the original purpose of this PR. I am closing this PR so that an appropriately described new PR can be created.
